plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/openai/OpenAICompletionEngine.java

```diff
@@ -16,396 +16,239 @@
-import com.theokanning.openai.completion.CompletionChoice;
-import com.theokanning.openai.completion.CompletionRequest;
-import com.theokanning.openai.completion.chat.ChatCompletionChoice;
 import com.theokanning.openai.completion.chat.ChatCompletionRequest;
+import com.theokanning.openai.completion.chat.ChatCompletionResult;
 import com.theokanning.openai.completion.chat.ChatMessage;
-import okhttp3.ResponseBody;
+import com.theokanning.openai.service.OpenAiService;
 import org.jkiss.code.NotNull;
-import org.jkiss.code.Nullable;
 import org.jkiss.dbeaver.DBException;
 import org.jkiss.dbeaver.Log;
-import org.jkiss.dbeaver.model.DBPDataSourceContainer;
 import org.jkiss.dbeaver.model.ai.AIConstants;
-import org.jkiss.dbeaver.model.ai.AIEngineSettings;
 import org.jkiss.dbeaver.model.ai.AISettingsRegistry;
-import org.jkiss.dbeaver.model.ai.completion.AbstractAICompletionEngine;
+import org.jkiss.dbeaver.model.ai.completion.DAIChatMessage;
+import org.jkiss.dbeaver.model.ai.completion.DAIChatRole;
 import org.jkiss.dbeaver.model.ai.completion.DAICompletionContext;
-import org.jkiss.dbeaver.model.ai.completion.DAICompletionMessage;
-import org.jkiss.dbeaver.model.ai.format.IAIFormatter;
-import org.jkiss.dbeaver.model.ai.metadata.MetadataProcessor;
-import org.jkiss.dbeaver.model.ai.openai.service.AdaptedOpenAiService;
-import org.jkiss.dbeaver.model.ai.openai.service.GPTCompletionAdapter;
-import org.jkiss.dbeaver.model.data.json.JSONUtils;
-import org.jkiss.dbeaver.model.exec.DBCExecutionContext;
+import org.jkiss.dbeaver.model.ai.completion.DAICompletionEngine;
+import org.jkiss.dbeaver.model.ai.n.AIStreamingResponseHandler;
+import org.jkiss.dbeaver.model.ai.n.DisposableLazyValue;
+import org.jkiss.dbeaver.model.data.DBDObject;
 import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
-import org.jkiss.dbeaver.model.struct.DBSObjectContainer;
-import org.jkiss.dbeaver.runtime.DBWorkbench;
-import org.jkiss.dbeaver.utils.RuntimeUtils;
-import org.jkiss.utils.CommonUtils;
 import retrofit2.HttpException;
-import retrofit2.Response;
 
-import java.io.IOException;
-import java.io.StringReader;
 import java.time.Duration;
-import java.util.*;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-import java.util.stream.Collectors;
+import java.util.List;
+import java.util.stream.Stream;
 
-public class OpenAICompletionEngine extends AbstractAICompletionEngine<GPTCompletionAdapter, Object> {
+public class OpenAICompletionEngine implements DAICompletionEngine {
     private static final Log log = Log.getLog(OpenAICompletionEngine.class);
 
-    //How many retries may be done if code 429 happens
-    protected static final int MAX_REQUEST_ATTEMPTS = 3;
-
-    private static final Map<String, GPTCompletionAdapter> clientInstances = new HashMap<>();
+    private static final int MAX_RETRIES = 3;
+    private static final Duration TIMEOUT = Duration.ofSeconds(30);
+    private static final String SYSTEM_PROMPT = """
```

Comment:
Currently system prompt is copy-pasted for every engine.
Let's move it to some middle level